icicle/appUtils/msm/msm.cu

```diff
@@ -132,32 +139,14 @@ namespace msm {
           buckets_indices[current_index] =
             (msm_index << (c + bm_bitsize)) | (bm << c) |
             bucket_index; // the bucket module number and the msm number are appended at the msbs
-          if (scalar == S::zero() || scalar == S::one() || bucket_index == 0)
-            buckets_indices[current_index] = 0; // will be skipped
-          point_indices[current_index] = tid;   // the point index is saved for later
+          if (scalar == S::zero() || bucket_index == 0) buckets_indices[current_index] = 0; // will be skipped
+          point_indices[current_index] = tid; // the point index is saved for later
 #endif
         }
       }
     }
 
-    template <typename P, typename A, typename S>
-    __global__ void
-    add_ones_kernel(A* points, S* scalars, P* results, const unsigned msm_size, const unsigned run_length)
-    {
-      unsigned tid = (blockIdx.x * blockDim.x) + threadIdx.x;
-      const unsigned nof_threads = (msm_size + run_length - 1) / run_length; // 129256
-      if (tid >= nof_threads) {
-        results[tid] = P::zero();
-        return;
-      }
-      const unsigned start_index = tid * run_length;
-      P sum = P::zero();
-      for (int i = start_index; i < min(start_index + run_length, msm_size); i++) {
-        if (scalars[i] == S::one()) sum = sum + points[i];
-      }
-      results[tid] = sum;
-    }
-
+    template <typename S>
```

Comment:
Do we want to use force_inline instead of unused typename params?
Same for `find_max_size` below